sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suite.scala

```diff
@@ -70,6 +73,34 @@ class HiveThriftServer2Suite extends FunSuite with Logging {
     port
   }
 
+  def withThriftUIAndJDBC(
+      uiPort: Int,
+      serverStartTimeout: FiniteDuration = 1.minute,
+      httpMode: Boolean = false)(
+      f: Statement => Unit) {
+    val port = randomListeningPort
+
+    startThriftServerWithUIPort(port, uiPort, serverStartTimeout, httpMode) {
+      val jdbcUri = if (httpMode) {
+        s"jdbc:hive2://${"localhost"}:$port/" +
+          "default?hive.server2.transport.mode=http;hive.server2.thrift.http.path=cliservice"
+      } else {
+        s"jdbc:hive2://${"localhost"}:$port/"
+      }
+
+      val user = System.getProperty("user.name")
+      val connection = DriverManager.getConnection(jdbcUri, user, "")
+      val statement = connection.createStatement()
+
+      try {
+        f(statement)
+      } finally {
+        statement.close()
+        connection.close()
+      }
+    }
+  }
+
   def withJdbcStatement(
```

Comment:
We can change `f: Statement => Unit` to `f: (Int, Int, Statement) => Unit`, and pass in server port and UI port. Then `withThriftUIAndJDBC` can be removed.